### PR TITLE
Add bracket-push exercise

### PR DESCRIPTION
**Pull request by jimmbraddock**

### CODE REVIEW

File: exercises/bracket-push/bracket_push_test.exs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+defmodule BracketPushTest do
+  use ExUnit.Case
+
+  # @tag :pending
+  test "checks for appropriate bracketing in a set of brackets" do
+    assert BracketPush.check_brackets("{}")
+  end
+
+  @tag :pending
+  test "test for unclosed brackets" do
```
Comment (parkerl) [inline exercises/bracket-push/bracket_push_test.exs]: As in rail fence, please update the test descriptions.

File: exercises/bracket-push/bracket_push_test.exs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    refute BracketPush.check_brackets("{{")
+  end
+
+  @tag :pending
+  test "test for more than one pair of brackets" do
+    assert BracketPush.check_brackets("{}[]")
+  end
+
+  @tag :pending
+  test "test fo brackets are out of order" do
```
Comment (parkerl) [inline exercises/bracket-push/bracket_push_test.exs]: typo

File: exercises/bracket-push/bracket_push_test.exs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  @tag :pending
+  test "checks bracket closure in a long string of brackets" do
+    assert BracketPush.check_brackets("{[]([()])}")
+  end 
+
+  @tag :pending
+  test "should ignore non-bracket characters" do
+    assert BracketPush.check_brackets("{hello[]([a()])b}c")
+  end
```
Comment (parkerl) [inline exercises/bracket-push/bracket_push_test.exs]: Add a test for an empty string, and a string with newlines?

File: exercises/bracket-push/example.exs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    |> String.replace(~r/[^\{\}\[\]\(\)]/, "")
+    |> String.codepoints
+    |> check([])
+  end
+
+  defp check([], []), do: true
+  defp check([], _), do: false
+  defp check([h|t], acc) do
+    cond do
+      Map.has_key?(@brackets, h) -> check(t, [Map.get(@brackets, h) | acc])
```
Comment (parkerl) [inline exercises/bracket-push/example.exs]: Please bring the result of each condition onto its own line and indent.

**Comment by parkerl** [comment]
@jimmbraddock thanks again for this exercise. In addition to the file comments, please scrub trailing whitespace from all files.

**Comment by parkerl** [comment]
Looking good! Thanks! :tada:
